### ISSUE TITLE
suggestion: point to specific version in rosinstall file (or tag repository)

### ISSUE THREAD

**Issue by gavanderhoorn** [ros-industrial-consortium/A5#16]
The [current a5.rosinstall](https://github.com/ros-industrial-consortium/A5/blob/821a8f423d1bb5ae494ad9e10a61f22ac488416d/a5.rosinstall) uses refs like `master` in the `version` field of some entries. While this is convenient during development, using such refs makes users checkout 'moving targets' when running `wstool` (`master` can be anything).

Would it be an idea to point entries (especially for external dependencies such as `pycom`, `o3d3xx` and `pepperl_fuchs`) to specific commit hashes? That way changes upstream don't potentially break the application as they will have to be tracked and tested explicitly.

**Comment by Jmeyer1292** [comment]
You're not wrong - it's probably a good idea for some of these items. On the other hand, the software is still under active development and I'd like to stay up to date. I upgrade and pull frequently on my machines.

Next time I'm down with the system, I'll pull the hashes and make a PR that changes this so we can better track "releases".

**Comment by gavanderhoorn** [comment]
@Jmeyer1292 wrote:
> On the other hand, the software is still under active development and I'd like to stay up to date. I upgrade and pull frequently on my machines.

yes, I understand. But are you also tracking changes for the external dependencies?

An alternative might be to have a `devel` branch (or similar) that has a `.rosinstall` tracking `HEAD` (or other named branches), and a `release` branch that targets specific commits. Or just a tag.
